inc/dashboardcards.class.php

```diff
@@ -0,0 +1,642 @@
+            'filters' => [
+                'credit','entity'
+            ]
+        ];
+
+        $cards["bn_percent_credit_used"] = [
+            'widgettype' => ["bigNumber"],
+            'itemtype' => "\\Credit",
+            'group' => __('Credit'),
+            'label' => __("Proportion of credit used"),
+            'provider' => PluginCreditDashboardCards::class . "::percentCreditsUsed",
+            'filters' => [
+                'credit'
+            ]
+        ];
+
+        $cards["date_end_ticket"] = [
+            'widgettype' => ["bigNumber"],
+            'itemtype' => "\\Credit",
+            'group' => __('Credit'),
+            'label' => __("End date of the contract"),
+            'provider' => PluginCreditDashboardCards::class . "::getDateEndingCredit",
+            'filters' => [
+                'credit'
+            ]
+        ];
+
+        $cards["credits_consumption_ticket"] = [
+            'widgettype' => ['bars'],
+            'itemtype' => "\\Credit",
+            'group' => __('Credit'),
+            'label' => __("Credit consumption by ticket"),
+            'provider' => PluginCreditDashboardCards::class . "::getCreditsConsumption",
+            'filters' => [
+                'credit'
+            ]
+        ];
+
+        $cards["credits_evolution_period"] = [
+            'widgettype' => ['lines', 'bars', 'areas'],
+            'itemtype' => "\\Credit",
+            'group' => __('Credit'),
+            'label' => __("Evolutions of credit consumption"),
+            'provider' => PluginCreditDashboardCards::class . "::getCreditsEvolution",
+            'filters' => [
+                'credit'
+            ]
+        ];
+
+        return $cards;
+    }
+
+    /**
+     * @param array $params
+     * @return array
+     */
+    public static function nbCredits(array $params = []): array
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $active_entity = Session::getActiveEntity();
+
+        $default_params = [
+            'label' => "",
+            'icon' => self::getIconWallet(),
+            'apply_filters' => [],
+            'alt' => 'initial quantity of credit'
+        ];
+        $params = array_merge($default_params, $params);
+
+        $t_table = PluginCreditEntity::getTable();
+
+        $criteria = array_merge_recursive(
+            [
+                'SELECT' => [
+                    'quantity'
+                ],
+                'FROM' => $t_table,
+                'WHERE' => [
+                    "$t_table.entities_id" => $active_entity
+                ]
+
+            ],
+            Provider::getFiltersCriteria($t_table, $params['apply_filters'])
+        );
+
+        $nb_items = 0;
+        foreach ($DB->request($criteria) as $id => $row) {
+            $nb_items += $row['quantity'];
+        }
+
+        return [
+            'number' => $nb_items,
+            'url' => '',
```

Comment:
A URL is required for the map to be clickable and open the corresponding search.

To be done on other cards too.

Reply:
> A URL is required for the card to be clickable and open the corresponding search.
> 
> To be done on other cards too.

after further research, an empty URL means that the widget is not clickable 

And this is legitimate in certain cases 

![image](https://github.com/user-attachments/assets/4fa4348b-d68b-4140-864e-d05449e81aaf)